src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionDescriptorBuilder.cs

```diff
@@ -739,11 +733,13 @@ private static bool IsAttributeRoutedAction(ControllerActionDescriptor actionDes
                 actionDescriptions.Add(description);
             }
 
+            var declaringType = actionDescriptor.MethodInfo.DeclaringType.GetTypeInfo();
             var methodFullName = string.Format(
                 CultureInfo.InvariantCulture,
-                "{0}.{1}",
-                actionDescriptor.MethodInfo.DeclaringType.FullName,
-                actionDescriptor.MethodInfo.Name);
+                "{0}.{1} ({2})",
+                declaringType.FullName,
+                actionDescriptor.MethodInfo.Name,
+                declaringType.Assembly.GetName().Name);
```

Comment:
Fair enough, I was actually going to comment that it was weird that we were using DeclaringType